lib/matplotlib/testing/compare.py

```diff
@@ -128,6 +133,109 @@ def convert(old, new):
+            else b"'" + b.replace(b"'", b"'\"'\"'") + b"'")
+
+
+class _SVGConverter(object):
+    def __init__(self):
+        self._proc = None
+        # We cannot rely on the GC to trigger `__del__` at exit because
+        # other modules (e.g. `subprocess`) may already have their globals
+        # set to `None`, which make `proc.communicate` or `proc.terminate`
+        # fail.  By relying on `atexit` we ensure the destructor runs before
+        # `None`-setting occurs.
+        atexit.register(self.__del__)
+
+    def _read_to_prompt(self):
+        """Did Inkscape reach the prompt without crashing?
+        """
+        stream = iter(functools.partial(self._proc.stdout.read, 1), b"")
+        prompt = (b"\n", b">")
+        n = len(prompt)
+        its = itertools.tee(stream, n)
+        for i, it in enumerate(its):
+            next(itertools.islice(it, i, i), None)  # Advance `it` by `i`.
+        while True:
+            window = tuple(map(next, its))
+            if len(window) != n:
+                # Ran out of data -- one of the `next(it)` raised
+                # StopIteration, so the tuple is shorter.
+                return False
+            if self._proc.poll() is not None:
+                # Inkscape exited.
+                return False
+            if window == prompt:
+                # Successfully read until prompt.
+                return True
+
+    def __call__(self, orig, dest):
+        if (not self._proc  # First run.
+                or self._proc.poll() is not None):  # Inkscape terminated.
+            env = os.environ.copy()
+            # If one passes e.g. a png file to Inkscape, it will try to
+            # query the user for conversion options via a GUI (even with
+            # `--without-gui`).  Unsetting `DISPLAY` prevents this (and causes
+            # GTK to crash and Inkscape to terminate, but that'll just be
+            # reported as a regular exception below).
+            env.pop("DISPLAY", None)  # May already be unset.
+            # Do not load any user options.
+            # `os.environ` needs native strings on Py2+Windows.
+            env[str("INKSCAPE_PROFILE_DIR")] = os.devnull
+            # Old versions of Inkscape (0.48.3.1, used on Travis as of now)
+            # seem to sometimes deadlock when stderr is redirected to a pipe,
+            # so we redirect it to a temporary file instead.  This is not
+            # necessary anymore as of Inkscape 0.92.1.
+            self._stderr = TemporaryFile()
+            self._proc = subprocess.Popen(
+                [str("inkscape"), "--without-gui", "--shell"],
+                stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                stderr=self._stderr, env=env)
+            if not self._read_to_prompt():
+                raise OSError("Failed to start Inkscape")
+
+        # Inkscape uses glib's `g_shell_parse_argv`, which has a consistent
+        # behavior across platforms, so we can just use `shlex.quote`.
```

Comment:
Yes. Moving it down.  Will also clarify the comment regarding `\n` (inkscape uses fgets so there's no choice).